src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const state2 = { ...state };
+  const arr = [];
+
+  for (const ch of actions) {
+    const { type, keysToRemove, extraData } = ch;
+    let state3 = {};
```

Comment:
The variable 'state3' is redundant, you can directly push the new state object into 'arr' after applying the changes.